src/test/java/org/broadinstitute/hellbender/tools/GatherVcfsCloudIntegrationTest.java

```diff
@@ -38,36 +49,40 @@
     }
 
     @Test(groups = "bucket", dataProvider = "getGatherTypes")
-    public void testGatherOverNIO(final GatherVcfsCloud.GatherType gatherType){
+    public void testGatherOverNIO(final GatherVcfsCloud.GatherType gatherType) throws IOException {
         assertGatherProducesCorrectVariants(gatherType,
                                             new File(largeFileTestDir, "1000G.phase3.broad.withGenotypes.chr20.10100000.vcf"),
                                             getTestFile("cloud-inputs.args"));
     }
 
     @Test(dataProvider = "getGatherTypes")
-    public void testLocalGathers(final GatherVcfsCloud.GatherType gatherType){
+    public void testLocalGathers(final GatherVcfsCloud.GatherType gatherType) throws IOException {
         assertGatherProducesCorrectVariants(gatherType,
                                             getTestFile("gzipped.vcf.gz"),
                                             getTestFile("bgzipped_shards.args"));
     }
 
-    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, File inputs) {
+    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, File inputs) throws IOException {
         assertGatherProducesCorrectVariants(gatherType, expected, Collections.singletonList(inputs));
     }
 
-    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, List<File> inputs) {
+    @SuppressWarnings({"unchecked"})
+    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, List<File> inputs) throws IOException {
         final File output = createTempFile("gathered", ".vcf.gz");
         final ArgumentsBuilder args = new ArgumentsBuilder();
         inputs.forEach(args::addInput);
         args.addOutput(output)
                 .addArgument(GatherVcfsCloud.GATHER_TYPE_LONG_NAME, gatherType.toString());
         runCommandLine(args);
 
-        try (  final FeatureDataSource<VariantContext> actualReader = new FeatureDataSource<>(output);
-               final FeatureDataSource<VariantContext> expectedReader = new FeatureDataSource<>(expected);){
+        try (final AbstractFeatureReader<? extends Feature, ?> actualReader = AbstractFeatureReader.getFeatureReader(output.getAbsolutePath(), null, FeatureManager.getCodecForFile(output, VariantContext.class), false, Function.identity(), Function.identity());
+             final AbstractFeatureReader<? extends Feature, ?> expectedReader = AbstractFeatureReader.getFeatureReader(expected.getAbsolutePath(), null, FeatureManager.getCodecForFile(expected, VariantContext.class), false, Function.identity(), Function.identity())){
```

Comment:
Type these as `FeatureReader<VariantContext>` instead. This should allow you to remove the gross typecasts below.

Reply:
I don't think I can type the feature readers that way without casting them by using AbstractFeatureReader.